src/main/java/org/opensearch/knn/indices/ModelGraveyard.java

```diff
@@ -151,27 +153,75 @@ public static NamedDiff readDiffFrom(StreamInput streamInput) throws IOException
      * @throws IOException
      */
     public static ModelGraveyard fromXContent(XContentParser xContentParser) throws IOException {
+        // Added validation checks to validate all the different possible scenarios
```

Comment:
Not sure if these lines of comments are any useful tbh. I think we just need a comment on what is the valid form for this method. All the invalid form can be validated in unit tests.